File: script_perf/factorial/python/factorial.py
#!/usr/bin/python

import sys

def factorial(n):
    ret = 1
    for i in range(1, n+1):
        ret = ret * i

    return ret


if len(sys.argv) < 2 :
    print "Please specify number which need calculate its factorial"
    exit(1)

num = int(sys.argv[1])
loops = 1
if len(sys.argv) >= 3:
    loops = int(sys.argv[2])

for i in range(0, loops):
    ret = factorial(num)

File: script_perf/factorial/php/factorial.php
#!/usr/bin/php
<?php

function factorial($n)
{
	if ($n == 0) {
		return 0;
	}

	$ret = 1;
	for ($i = 1; $i <= $n; ++$i) {
		$ret = $ret * $i;
	}

	return $ret;
}


if ($argc < 2) {
	echo "Please specify the number which need calculate its factorial\n";
	return 1;
}

$num = $argv[1];
$loops = 1;

if ($argc == 3) {
	$loops = $argv[2];
}

for ($i = 0; $i < $loops; ++$i) {
	$ret = factorial($num);
}

echo "$ret\n";
?>
